pkg/cable/vxlan/vxlan.go

```diff
@@ -0,0 +1,538 @@
+	}
+
+	v.connections = append(v.connections, v1.Connection{Endpoint: remoteEndpoint.Spec, Status: v1.Connected,
+		UsingIP: endpointInfo.UseIP, UsingNAT: endpointInfo.UseNAT})
+
+	klog.V(log.DEBUG).Infof("Done adding endpoint for cluster %s", remoteEndpoint.Spec.ClusterID)
+
+	return endpointInfo.UseIP, nil
+}
+
+func (v *vxLan) DisconnectFromEndpoint(remoteEndpoint types.SubmarinerEndpoint) error {
+	klog.V(log.DEBUG).Infof("Removing endpoint %#v", remoteEndpoint)
+
+	if v.localEndpoint.Spec.ClusterID == remoteEndpoint.Spec.ClusterID {
+		klog.V(log.DEBUG).Infof("Will not disconnect self")
+		return nil
+	}
+
+	v.mutex.Lock()
+	defer v.mutex.Unlock()
+
+	// parse remote addresses and allowed IPs
+	ip := submarinerEndpointIP(&remoteEndpoint)
+
+	remoteIP := net.ParseIP(ip)
+
+	if remoteIP == nil {
+		return fmt.Errorf("failed to parse remote IP %s", ip)
+	}
+
+	allowedIPs := parseSubnets(remoteEndpoint.Spec.Subnets)
+
+	err := v.vxLanIface.DelFDB(remoteIP, "00:00:00:00:00:00")
+
+	if err != nil {
+		return fmt.Errorf("failed to delete remoteIP %q from the forwarding database", remoteIP)
+	}
+
+	err = v.vxLanIface.DelRoute(allowedIPs)
+
+	if err != nil {
+		return fmt.Errorf("failed to remove route for the CIDR %q",
+			allowedIPs)
+	}
+
+	v.connections = removeConnectionForEndpoint(v.connections, remoteEndpoint)
+	cable.RecordDisconnected(CableDriverName, &v.localEndpoint.Spec, &remoteEndpoint.Spec)
+
+	klog.V(log.DEBUG).Infof("Done removing endpoint for cluster %s", remoteEndpoint.Spec.ClusterID)
+
+	return nil
+}
+
+func removeConnectionForEndpoint(connections []v1.Connection, endpoint types.SubmarinerEndpoint) []v1.Connection {
+	for j := range connections {
+		if connections[j].Endpoint.CableName == endpoint.Spec.CableName {
+			copy(connections[j:], connections[j+1:])
+			return connections[:len(connections)-1]
+		}
+	}
+
+	return connections
+}
+
+func (v *vxLan) GetConnections() ([]v1.Connection, error) {
+	return v.connections, nil
+}
+
+func (v *vxLan) GetActiveConnections() ([]v1.Connection, error) {
+	return v.connections, nil
+}
+
+func (iface *vxLanIface) AddFDB(ipAddress net.IP, hwAddr string) error {
+	macAddr, err := net.ParseMAC(hwAddr)
+
+	if err != nil {
+		return fmt.Errorf("invalid MAC Address (%s) supplied. %v", hwAddr, err)
+	}
+
+	if ipAddress == nil {
+		return fmt.Errorf("invalid ipAddress (%v) supplied", ipAddress)
+	}
+
+	neigh := &netlink.Neigh{
+		LinkIndex:    iface.link.Index,
+		Family:       unix.AF_BRIDGE,
+		Flags:        netlink.NTF_SELF,
+		Type:         netlink.NDA_DST,
+		IP:           ipAddress,
+		State:        netlink.NUD_PERMANENT | netlink.NUD_NOARP,
+		HardwareAddr: macAddr,
+	}
+
+	err = netlink.NeighAppend(neigh)
+	if err != nil {
+		return fmt.Errorf("unable to add the bridge fdb entry %v, err: %s", neigh, err)
+	} else {
+		klog.V(log.DEBUG).Infof("Successfully added the bridge fdb entry %v", neigh)
+	}
+
+	return nil
+}
+
+func (iface *vxLanIface) DelFDB(ipAddress net.IP, hwAddr string) error {
```

Comment:
It's recommended to keep receiver var names to 1 or 2 chars (I thought we had linting that caught this). 
 
```suggestion
func (v *vxLanIface) DelFDB(ipAddress net.IP, hwAddr string) error {
```